normandy/recipes/exports.py

```diff
@@ -0,0 +1,120 @@
+import logging
+
+import kinto_http
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+
+logger = logging.getLogger(__name__)
+
+
+def recipe_as_record(recipe):
+    """
+    Transform a recipe to a dict with the minimum amount of fields needed for clients
+    to verify and execute recipes.
+
+    :param recipe: a recipe ready to be exported.
+    :returns: a dict to be posted on Remote Settings.
+    """
+    from normandy.recipes.api.v1.serializers import (
+        MinimalRecipeSerializer,
+    )  # avoid circular imports
+
+    serializer = MinimalRecipeSerializer(recipe)
+    record = serializer.data
+    record["id"] = str(recipe.id)
+    return record
+
+
+class RemoteSettings:
+    """
+    Interacts with a RemoteSettings service.
+
+    Basically, a recipe becomes a record in the dedicated collection on Remote Settings.
+    When it is disabled, the associated record is deleted.
```

Comment:
Will Remote Settings mind if we create a recipe, enable it, disable, and then re-enable it again? Does Remote Settings have the concept of undeleting a record? Or will it create a new record?

Reply:
When the Remote Settings client fetches the changes from the server, there is no distinction between a record creation from update. 

If the client had sync and caught the deletion, it will be interpreted as a creation. If the client hadn't caught the deletion, the re-creation will be interpreted as an update.

Does it make sense?

Reply:
I'm less concerned about the client and more about the server. Since I know that kinto uses tombstones for deleted records, I wanted to check that the server won't mind if we try and resurrect a tombstone. It sounds like you're not worried though, so I expect we're fine here.